File: API/Entities/Service.cs
using System;
using System.Collections;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Service
{
    public int Id { get; set; }
    public string Name { get; set; } = "";
    public decimal Price { get; set; } = 0;
    public string Description { get; set; } = ""; 
    public byte DurationInMinutes { get; set; } = 0;
    public bool IsAvailable { get; set; } = true;  
    public DateTime Created { get; set; } = DateTime.UtcNow;

    public ICollection<Photo> Photos { get; set; } = [];

    public ICollection<Appointment> Appointments { get; set; } = [];

    public ICollection<Review> Reviews { get; set; } = [];

}

File: Frontend/src/app/types/service.ts
import type { Review } from './review';

export interface Service {
  id: number;
  name: string;
  price: number;
  description: string;
  durationInMinutes: number;
  isAvailable: boolean;
  created: string;
  mainPhotoUrl?: string | null;
  photos: Photo[];
}

export interface Photo {
  id: number;
  url: string;
  isMain: boolean;
}

export interface ServiceDetail extends Service {
  reviews: Review[];
}

export interface ServiceQueryParams {
  pageNumber?: number;
  pageSize?: number;
  search?: string;
  isAvailable?: boolean | null;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
}

export interface ServiceFormValue {
  name: string;
  price: number;
  description: string;
  durationInMinutes: number;
  isAvailable: boolean;
}
